plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts

```diff
@@ -257,36 +249,28 @@ export class ArgoService implements ArgoServiceApi {
       },
     };
 
-    const options: AxiosRequestConfig = {
+    const options: RequestInit = {
       method: 'POST',
-      url: `${baseUrl}/api/v1/projects`,
       headers: {
         'Content-Type': 'application/json',
         Authorization: `Bearer ${argoToken}`,
       },
-      data: data,
+      body: JSON.stringify(data),
     };
-    try {
-      const resp = await axios.request(options);
-      return resp.data;
-    } catch (error: any) {
-      console.error(error);
-      if (error.response?.status === 404) {
-        return error.response?.data;
-      } else if (error.response?.status === 403) {
-        throw new Error(error.response?.data.message);
-      } else if (
-        error.response?.data.message.includes(
-          'existing project spec is different',
-        )
-      ) {
-        throw new Error(
-          'Duplicate project detected. Cannot overwrite existing.',
-        );
-      } else {
-        throw new Error('Could not create ArgoCD project.');
-      }
+    const resp = await fetch(`${baseUrl}/api/v1/projects`, options);
+    const responseData = await resp.json();
+    if (resp.status === 403 || resp.status === 404) {
+      throw new Error(responseData.message);
```

Comment:
We will throw now in cases when it is 404. In the previous version we did return when we got 404. Can you double check it will be ok to throw an Error in case of 404 ?

Reply:
Change it back to return the data.